xarray/tests/test_backends.py

```diff
@@ -3012,15 +3012,12 @@ def gen_datasets_with_common_coord_and_time(self):
 
         return ds1, ds2
 
-    @pytest.mark.parametrize("combine", ["nested", "by_coords"])
     @pytest.mark.parametrize("opt", ["all", "minimal", "different"])
     @pytest.mark.parametrize("join", ["outer", "inner", "left", "right"])
-    def test_open_mfdataset_does_same_as_concat(self, combine, opt, join):
+    def test_open_mfdataset_does_same_as_concat(self, opt, join):
         with self.setup_files_and_datasets() as (files, [ds1, ds2]):
-            if combine == "by_coords":
-                files.reverse()
             with open_mfdataset(
-                files, data_vars=opt, combine=combine, concat_dim="t", join=join
+                files, data_vars=opt, combine="nested", concat_dim="t", join=join
```

Comment:
does `join` make sense for `combine="by_coords"`. If so, we should test it with

```
@pytest.mark.parametrize("combine, concat_dim", [("nested", "t"), ("by_coords", None)])
```

Reply:
I think it does - I've reinstated these tests, just with `concat_dim=None`. I think this is done now.